methods/add_books.rb

```diff
@@ -0,0 +1,16 @@
+require_relative '../book'
+require_relative './books_memory'
+
+class AddBooks
+  include BooksMemory
+
+  def add
+    print "\nTitle: "
+    title = gets.chomp.capitalize
+    print 'Author: '
+    author = gets.chomp.capitalize
```

Comment:
- [ ] Your `add` method here prints information, gets user inputs, and also saves the inputs. This violates the [Single-responsibility](https://www.digitalocean.com/community/conceptual_articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design#single-responsibility-principle) rule of SOLID. The sole purpose of the `add` method should be to add a book. Outputs and users' inputs should not be its concern. Kindly extract the requests for input and input validation to another method/class whose responsibilities are to request data.

  **SUGGESTION:** You could extract the requests for input into another method and have that method return an array of the `title` and `author` variables (see the first image). Then in add method, you can use destructuring to obtain those variables (see the second image)
  
  ![alex-book-details](https://user-images.githubusercontent.com/36542986/152005591-2a3fa4f9-c710-4f45-8517-59ed0765b3ad.jpg)
  ![alex-book-used](https://user-images.githubusercontent.com/36542986/152005600-5c9ffaa7-a6bb-4f97-8fe7-21ddeaf6d049.jpg)

Reply:
Hi @TedLivist  thanks for your suggestions, I have one question, do the SOLID rules refers to the methods? I mean one method must do only one task? and can I have as many methods as I need inside a class? is that rigth?

Reply:
Hi @AlexRS90, the point of SOLID rules is to ensure best practices. Now, while the SOLID terminology often mentions 'classes' specifically, the rules are often implemented at the method levels too.

Consider that it makes more sense to have two methods - each with a single responsibility - than to have one method performing two responsibilities.

Yeah. You can have any number methods in your classes as you deem fit.